packages/frontend/components/api/RequestBuilder.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import {
+  Button,
+  Input,
+  Text,
+  Tooltip,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+} from '@chakra-ui/react'
+import { BigNumber } from 'ethers'
+import { formatFixed } from '@ethersproject/bignumber'
+import { useContractFunction, TransactionState } from '@usedapp/core'
+import { ContractId } from '../../conf/config'
+import { Error } from '../../components/Error'
+import { useContract } from '../../hooks/useContract'
+// @ts-ignore
+import { APIRequestBuilder } from 'types/typechain'
+
+const DEFAULT_MULTIPLIER = '1000000000000000000'
+const DEFAULT_URL =
+  'https://min-api.cryptocompare.com/data/pricemultifull?fsyms=ETH&tsyms=USD'
+const DEFAULT_PATH = 'RAW.ETH.USD.VOLUME24HOUR'
+const URL_REGEX =
+  /^(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/
+const PATH_REGEX = /^[a-zA-Z_][\w]*(?:\.[\w]+)*$/
+const MULTIPLIER_REGEX = /^(1(0)*)$/
+
+export const RequestBuilder = (): JSX.Element => {
+  const [url, setURL] = useState(DEFAULT_URL)
+  const [path, setPath] = useState(DEFAULT_PATH)
+  const [multiplier, setMultiplier] = useState(DEFAULT_MULTIPLIER)
+  const [requestId, setRequestId] = useState('')
+  const [data, setData] = useState('')
+
+  const apiRequestBuilder = useContract<APIRequestBuilder>(
+    ContractId.APIRequestBuilder
+  )
+
+  const { send, state, events } = useContractFunction(
+    apiRequestBuilder,
+    'requestData',
+    { transactionName: 'External API Request' }
+  )
+
+  const requestData = useCallback(async () => {
+    setData('')
+    setRequestId('')
+
+    await send(url, path, BigNumber.from(multiplier))
+    setData(null)
+  }, [send, url, path, multiplier])
+
+  const readData = useCallback(async () => {
+    const res = (await apiRequestBuilder.data()).toString()
```

Comment:
```suggestion
    const res = await apiRequestBuilder.data()
```

`formatFixed` accepts `BigNumberish`.